File: backend/app/models/message.py
"""
消息数据模型
"""

from sqlalchemy import Column, Integer, String, DateTime, Boolean, Text, ForeignKey, Enum
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base
import enum


class MessageType(enum.Enum):
    """消息类型枚举"""
    USER = "user"
    PET = "pet"
    SYSTEM = "system"


class Message(Base):
    """消息模型"""
    __tablename__ = "messages"
    
    id = Column(Integer, primary_key=True, index=True)
    conversation_id = Column(String(100), nullable=False, index=True)  # 会话ID
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    pet_id = Column(Integer, ForeignKey("pets.id"), nullable=False)
    message_type = Column(Enum(MessageType), nullable=False)
    content = Column(Text, nullable=False)  # 消息内容
    extra_data = Column(Text, nullable=True)  # 附加信息（JSON字符串）
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # 关系
    user = relationship("User")
    pet = relationship("Pet") 

File: FurryKids/FurryKids/Models/Message.swift
import Foundation

// 注意：这些类型已在FurryKidsApp.swift中定义，这里保留示例数据

// 示例数据
extension Message {
    static let samples = [
        Message(
            content: "汪汪！主人你回来啦！我好想你呀~ 🐕💕",
            type: .pet,
            mood: "兴奋"
        ),
        Message(
            content: "我也很想你！今天乖不乖？",
            type: .user
        ),
        Message(
            content: "我超级乖的！一直在等你回家，还自己玩了一会儿球球。不过有点想出去散步...",
            type: .pet,
            mood: "期待"
        )
    ]
} 